WordPress/src/main/java/org/wordpress/android/ui/me/MyProfileActivity.java

```diff
@@ -0,0 +1,143 @@
+    private WPTextView mAboutMe;
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        AccountHelper.getDefaultAccount().fetchAccountSettings(new ApiHelper.GenericCallback() {
+            @Override
+            public void onSuccess() {
+                refreshDetails();
+            }
+
+            @Override
+            public void onFailure(ApiHelper.ErrorType errorType, String errorMessage, Throwable throwable) {
+            }
+        });
+
+        ActionBar actionBar = getSupportActionBar();
+        if (actionBar != null) {
+            actionBar.setHomeButtonEnabled(true);
+            actionBar.setDisplayHomeAsUpEnabled(true);
+        }
+        setContentView(R.layout.my_profile_activity);
+
+        setTitle(R.string.my_profile);
+
+        mFirstName = (WPTextView) findViewById(R.id.first_name);
+        mLastName = (WPTextView) findViewById(R.id.last_name);
+        mDisplayName = (WPTextView) findViewById(R.id.display_name);
+        mAboutMe = (WPTextView) findViewById(R.id.about_me);
+
+        refreshDetails();
+
+        findViewById(R.id.first_name_row).setOnClickListener(createOnClickListener(getString(org.wordpress.android.R.string.first_name), null, mFirstName));
+        findViewById(R.id.last_name_row).setOnClickListener(createOnClickListener(getString(R.string.last_name), null, mLastName));
+        findViewById(R.id.display_name_row).setOnClickListener(createOnClickListener(getString(R.string.public_display_name), null, mDisplayName));
+        findViewById(R.id.about_me_row).setOnClickListener(createOnClickListener(getString(R.string.about_me), getString(R.string.about_me_hint), mAboutMe));
+    }
+
+    @Override
+    public void finish() {
+        super.finish();
+        ActivityLauncher.slideOutToRight(this);
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(final MenuItem item) {
+        if (item.getItemId() == android.R.id.home) {
+            onBackPressed();
+            return true;
+        }
+        return super.onOptionsItemSelected(item);
+    }
+
+    private void refreshDetails() {
+        Account account = AccountHelper.getDefaultAccount();
+        updateLabel(mFirstName, account != null ? StringUtils.unescapeHTML(account.getFirstName()) : null);
+        updateLabel(mLastName, account != null ? StringUtils.unescapeHTML(account.getLastName()) : null);
+        updateLabel(mDisplayName, account != null ? StringUtils.unescapeHTML(account.getDisplayName()) : null);
+        updateLabel(mAboutMe, account != null ? StringUtils.unescapeHTML(account.getAboutMe()) : null);
+    }
+
+    private void updateMyProfileForLabel(TextView textView) {
+        Map<String, String> params = new HashMap<>();
+        params.put(restParamForTextView(textView), textView.getText().toString());
+        AccountHelper.getDefaultAccount().postAccountSettings(params, new ApiHelper.GenericCallback() {
```

Comment:
I'm wondering if users who quickly modify several entries over a slow connection will run into problems. Just a thought, but perhaps instead of saving each entry as it's changed (which I initially suggested) we could save everything all at once when the user leaves the activity?